### ISSUE TITLE
Enable source-build pre-built detection

### ISSUE THREAD

**Issue by oleksandr-didyk** [dotnet/msbuild#8382]
Part of https://github.com/dotnet/source-build/issues/3017

Enable source-build pre-build detection on the current repository and resolve any pre-build issues discovered

**Comment by oleksandr-didyk** [comment]
As part of work on the task we discovered that in the context of repo build `msbuild`'s source-build is utilizing older versions of several components (for example, 7.0 and 6.0 versions of system libraries). In the context of current product source-build, these dependencies will be replaced by their respective latest versions. This creates a sizable difference in behaviour between the product source-build and the repo source-build.

@rainersigwald  pinging you here to get an opinion on what, in your mind, is a better option from `msbuild`'s perspective -> to have source-build continue using the latest versions (i.e. source-build differs from 'VS build') or for source-build to use the VS-compatible ones (i.e. source-build behaves the same as 'VS build')? The later will become possible with the introduction of repository `ProjectVersions.props` files.

CC: @MichaelSimons @mmitche

**Comment by rainersigwald** [comment]
I thought we had no choice but the former. For .NET SDK scenarios we have no problem with the idea of using the latest everything--that's what we expect at runtime. For Visual Studio, we must move at the pace of Visual Studio and be on stable releases.

There's some risk of churn/annoyances with building against two different versions, but I'm willing to give it a try and complain if it becomes difficult.

**Comment by tkapin** [comment]
@oleksandr-didyk - should this be closed already?

**Comment by oleksandr-didyk** [comment]
> @oleksandr-didyk - should this be closed already?

Yeah, the issue wasn't linked to the resolving PR - https://github.com/dotnet/msbuild/pull/8678

Closing
